### ISSUE TITLE
Skip building models that take more than a prespecified time to execute

### ISSUE THREAD

**Issue by vivekmishra369** [shankarpandala/lazypredict#65]
I would like to be able to pass an argument called timeout which skips a model if building the models crosses the timeout seconds

**Comment by devdjdjdj** [comment]
@shankarpandala  What help do you need here ?

**Comment by shankarpandala** [comment]
> @shankarpandala  What help do you need here ?

Do you have any idea to implement this?
If you like to implement this go ahead and raise pull request to dev branch once completed
I tried this and it was able to do only for linux.

**Comment by devdjdjdj** [comment]
> Do you have any idea to implement this?

The basic idea would be to track the execution time and halt the subprocess at timeout.

> I tried this and it was able to do only for linux.

Was the successful implementation in linux using signal ?
If not, then [this](https://stackoverflow.com/questions/366682/how-to-limit-execution-time-of-a-function-call-in-python) or [this](https://stackoverflow.com/questions/492519/timeout-on-a-function-call) should help.

**Comment by shankarpandala** [comment]
> > Do you have any idea to implement this?
> 
> The basic idea would be to track the execution time and halt the subprocess at timeout.
> 
> > I tried this and it was able to do only for linux.
> 
> Was the successful implementation in linux using signal ?
> If not, then [this](https://stackoverflow.com/questions/366682/how-to-limit-execution-time-of-a-function-call-in-python) or [this](https://stackoverflow.com/questions/492519/timeout-on-a-function-call) should help.
> 
> 
> 
> 

Yes I was able to do it using signal in linux but it is not working in windows

**Comment by shankarpandala** [comment]
This feature has been IMPLEMENTED in PR #539! ���

**Solution:**
Use the new `timeout` parameter to skip models that take too long:

```python
from lazypredict.Supervised import LazyClassifier, LazyRegressor

# Skip models that take more than 60 seconds
clf = LazyClassifier(timeout=60, verbose=1)
models, predictions = clf.fit(X_train, X_test, y_train, y_test)

# For regression
reg = LazyRegressor(timeout=120)  # 2 minute timeout
```

**How it works:**
- Models that exceed the timeout are automatically skipped
- A warning is printed (if verbose > 0) indicating which model timed out
- Other models continue to run normally
- Prevents indefinite waiting on slow models

Closing as COMPLETED - this exact feature is now available in the latest version!
